src/unitxt/operators.py

```diff
@@ -1293,6 +1313,52 @@ def _is_required(self, instance: dict) -> bool:
         return True
 
 
+class FilterByConditionBasedOnFields(FilterByCondition):
+    """Filters a stream, yielding only instances in which the values in required fields follow the required condition operator. The value is another field in the dataset.
```

Comment:
The description not clear (and seems to by mostly copied from the FilterByConditition).

Reply:
improved it a bit